### PR TITLE
Add Node Maintenance Operator to Medik8s

### PR DESCRIPTION
**Pull request by razo7**
We have moved Node Maintenance Operator (NMO) from KubeVirt to Medik8s, thus a proper documentation for NMO has to be created in Medik8s.io.
This PR is the first step for better presentation of NMO in Medik8s docs with the adding of NMO page and icon. 
https://issues.redhat.com/browse/ECOPROJECT-610

### CODE REVIEW

File: maintenance-node.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+<img src="../images/operator-icon/nmo_blue_icon.png" alt="nmo-icon" width="150" style="margin-left:auto; margin-right:auto; display:block"/>
+
+## The Problem
+
+Nodes in a Kubernetes cluster (especially bare meteal nodes) face many problems, and they can fail due to a kernel error or a NIC card hardware failure. The workloads on the failed node need to be restarted on another node in the cluster while the problem node is repaired or replaced. Node maintenance mode allows cluster administrators to gracefully turn-off nodes, move workloads to other parts of the cluster, and ensure that workloads do not get interrupted. Detailed progress and node status details are provided during maintenance.
```
Comment (slintes) [inline maintenance-node.md]: typo: bare meteal nodes

> Node maintenance mode allows...

- isn't this the solution, and not the problem (see header)?
- things lister after that are in wrong order (turn off, move ...)
- turning off is not part of NMO, here it sounds to me like it is

also: please wrap long lines

File: maintenance-node.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Maintenance Mode
+Node Maintenance Operator (NMO) is an open soucrce Kubernetes operator which keep nodes cordoned and drained while a matching NodeMaintenance (nm) custom resource (CR) exists.
```
Comment (slintes) [inline maintenance-node.md]: keep**s**

Maybe explain what cordon and drain means?
Maybe mention that this is a declarative way for `kubectl cordon / drain`?

File: maintenance-node.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+1. TOC
+{:toc}
+
+<img src="../images/operator-icon/nmo_blue_icon.png" alt="nmo-icon" width="150" style="margin-left:auto; margin-right:auto; display:block"/>
+
+## The Problem
+
+Nodes in a Kubernetes cluster (especially bare meteal nodes) face many problems, and they can fail due to a kernel error or a NIC card hardware failure. The workloads on the failed node need to be restarted on another node in the cluster while the problem node is repaired or replaced. Node maintenance mode allows cluster administrators to gracefully turn-off nodes, move workloads to other parts of the cluster, and ensure that workloads do not get interrupted. Detailed progress and node status details are provided during maintenance.
+
+## Maintenance Mode
```
Comment (slintes) [inline maintenance-node.md]: The solution? 🙂

File: maintenance-node.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+Node Maintenance Operator (NMO) is an open soucrce Kubernetes operator which keep nodes cordoned and drained while a matching NodeMaintenance (nm) custom resource (CR) exists.
+
+The operator watches for new or deleted CRs which indicate that a node in the cluster should move into maintenance when nm CR has been created or outisde of maintenance when nm CR has been deleted.
```
Comment (slintes) [inline maintenance-node.md]: not sure if "move" is the right wird here. OCP docs use "place".
So maybe replace "should move into maintenance when nm CR has been created or outisde of maintenance" with "should be placed into maintenance when nm CR has been created, or should end maintenance..."?

File: maintenance-node.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+When the node is in maintanince mode, then it is cordoned - set as unschedulable, and all the (possible) pods are evicted from that node.
```
Comment (slintes) [inline maintenance-node.md]: cordon is set unschedulable only
eviction is drain 🙂

File: maintenance-node.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+When the node leaves maintenance mode, then it is uncordoned - set as schedulable, and the NMO pod is removed.
```
Comment (slintes) [inline maintenance-node.md]: which pod is removed? 🤔

File: maintenance-node.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+Previously NMO was developed and bundled in KubeVirt project (or [OpenShift Virtualization](https://docs.openshift.com/container-platform/latest/virt/about-virt.html)).
```
Comment (slintes) [inline maintenance-node.md]: - don't use anything downstream in upstream docs
- not sure if we need to mention this at all 🤔  I'm fine without

File: maintenance-node.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+
+## Usage
+The operator can be installed using OperatorHub or [built and run from source](https://github.com/medik8s/node-maintenance-operator#build-and-run-the-operator).
```
Comment (slintes) [inline maintenance-node.md]: what abou adding a link to operatorhub.io?

File: maintenance-node.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+See in [here](https://github.com/medik8s/node-maintenance-operator#setting-node-maintenance) how to set/unset the node into maintenance mode, and also to understand [nm CR status](https://github.com/medik8s/node-maintenance-operator#nodemaintenance-status).
```
Comment (slintes) [inline maintenance-node.md]: "here" links are not nice. What about: "See [NMO's Readme]..."

File: maintenance-node.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+In addition, OpenShift users can use NMO directly from the Red Hat catalog (please see Node Maintenance Operator documnetaion in [OpenShfit Container Platform (OCP)](https://docs.openshift.com/container-platform/latest/nodes/nodes/eco-node-maintenance-operator.html)).
```
Comment (slintes) [inline maintenance-node.md]: @beekhof what is our policy with regards to mentioning downstream (OCP) in upstream docs. I'm used from KubeVirt times that downstream should noy be mentioned.

**Review by slintes — CHANGES_REQUESTED** [review]
Left some comments...

File: maintenance-node.md (hunk 4 above)
Comment (razo7) [reply maintenance-node.md]: Good point

File: maintenance-node.md (hunk 6 above)
Comment (razo7) [reply maintenance-node.md]: I was thinking about removing the CR, not the pod. I have deleted it.

File: maintenance-node.md (hunk 5 above)
Comment (razo7) [reply maintenance-node.md]: Do you meant it wasn't clear before?

File: maintenance-node.md (hunk 1 above)
Comment (razo7) [reply maintenance-node.md:19]: > typo: bare meteal nodes

I don't understand what did you mean by writing this.

File: maintenance-node.md (hunk 7 above)
Comment (razo7) [reply maintenance-node.md]: After talking with Andrew we should avoid mentioning downstream..

File: maintenance-node.md (hunk 10 above)
Comment (razo7) [reply maintenance-node.md]: After talking with Andrew we should avoid mentioning downstream..

File: maintenance-node.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+<img src="../images/operator-icon/nmo_blue_icon.png" alt="nmo-icon" width="150" style="margin-left:auto; margin-right:auto; display:block"/>
+
+## The Problem
+
+Nodes in a Kubernetes cluster (especially bare metal nodes) face many problems, and they can fail due to a kernel error or a NIC card hardware failure.
+The workloads on the failed node need to be restarted on another node in the cluster while the problem node is repaired or replaced.
+
+## The Solution
+Node Maintenance Operator (NMO) is an open soucrce Kubernetes operator which keeps nodes cordoned and drained while a matching NodeMaintenance (nm) custom resource (CR) exists.
```
Comment (mshitrit) [inline maintenance-node.md]: Typo, `soucrce` instead of `source`

File: maintenance-node.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+The operator perform a declarative way for doing [`kubectl cordon NODE`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#cordon) (set node as unschedulable), 
+[`kubectl drain NODE`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#drain) (evict pods from node), and [`kubectl taint NODE NAME KEY_1=VAL_1:TAINT_EFFECT_1`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#taint) (add taint for node).
+The operator watches for new or deleted CRs which indicate that a node in the cluster should be placed into maintenance when nm CR has been created, or should end maintenance when nm CR has been deleted.
+When the node is in maintanince mode, then it is cordoned - set as unschedulable, and all the (possible) pods are drained (evicted) from that node.
```
Comment (mshitrit) [inline maintenance-node.md]: Typo `maintanince` instead of `maintenance`

File: maintenance-node.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+When the node leaves maintenance mode, then it is uncordoned - set as schedulable.
+Detailed progress and node status details are provided during maintenance.
+
+## Usage
+The operator can be installed using [OperatorHub](https://operatorhub.io/operator/node-maintenance-operator) or [built and run from source](https://github.com/medik8s/node-maintenance-operator#build-and-run-the-operator).
```
Comment (mshitrit) [inline maintenance-node.md]: OperatorHub link seems to be broken

File: maintenance-node.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+See [NMO's Readme](https://github.com/medik8s/node-maintenance-operator#setting-node-maintenance) for how to set/unset the node into maintenance mode, and also for better understanding the  [nm CR status](https://github.com/medik8s/node-maintenance-operator#nodemaintenance-status).
```
Comment (mshitrit) [inline maintenance-node.md]: NIT: I could be wrong, but I think that: "better understanding `of` the ..."  works better

File: maintenance-node.md (hunk 13 above)
Comment (razo7) [reply maintenance-node.md]: Yes, I was thinking that this [PR](https://github.com/k8s-operatorhub/community-operators/pull/1624) will be merged already...

**Comment by razo7** [comment]
Since the first page is mainly about fencing, and the NMO specifically is not part of that solution/flow, thus we need extra thought or restructuring before we include in the front page.

**Pull request merged**